From b13708039a031c12bb42995e06e4149165dc76c4 Mon Sep 17 00:00:00 2001
From: "yi.zhang1" <yi.zhang1@amlogic.com>
Date: Fri, 19 May 2023 11:47:55 +0800
Subject: [PATCH] amlbian: amlbian uboot clean fail  [1/1]

PD#SWPL-123654

Problem:
amlbian uboot clean fail.

Solution:
fix this uboot clean case.

Verify:
local

Signed-off-by: yi.zhang1 <yi.zhang1@amlogic.com>
Change-Id: I07f3cebdcb7bce24094b497e8a7c3494699846e6
---
 config/functions/build-uboot | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/config/functions/build-uboot b/config/functions/build-uboot
index 9000c91..7c6f257 100755
--- a/config/functions/build-uboot
+++ b/config/functions/build-uboot
@@ -178,15 +178,10 @@ build_uboot_deb() {
 clean_uboot() {
 	for UBOOT_DIR in $UBOOT_DIR; do break; done
 	info_msg "Clean u-boot..."
-	[ -e "$UBOOT_DIR/Makefile" ] || return 0
-	
-	echo "make distclean -C"$UBOOT_DIR""
-	make distclean -C"$UBOOT_DIR"
-
-	rm $LINUX_DIR/.buildpass
+	rm -rf $UBOOT_DIR
 
 	# remove debs
-	if [[ ! -f $BUILD_DEBS/$VERSION/$WORK_BOARD/${LINUX_UBOOT_DEB}_${VERSION}-${UBOOT_VER}_${DISTRIB_ARCH}.deb ]]; then
+	if [ -f $BUILD_DEBS/$VERSION/$WORK_BOARD/${LINUX_UBOOT_DEB}_${VERSION}-${UBOOT_VER}_${DISTRIB_ARCH}.deb ]; then
 		rm $BUILD_DEBS/$VERSION/$WORK_BOARD/${LINUX_UBOOT_DEB}_${VERSION}-${UBOOT_VER}_${DISTRIB_ARCH}.deb
 	fi
 }
-- 
2.25.1

